### DISCUSSION TITLE
Add emojis to password

### DISCUSSION THREAD

**Discussion by farshadb** [openssl/openssl#22847] [Q&A]
Is there any way to create password including emojis?

**Comment by nhorman** [comment]
Most apis requiring a password to be set (PEM_write_PrivateKey for instance), allow the setting of a password callback.  Those callbacks accept a char *buffer, so a UTF16 emoji can't be specified directly, but theres nothing stopping you from writing a password callback that prompts the user for a password which does accept UTF16, and translates/hashes that to a larger array of unsigned chars.

**Comment by jamuir** [comment]
Also, if you are interested mainly in the "openssl" command line utility, then it is possible to read the password from a file as described here:

https://www.openssl.org/docs/manmaster/man1/openssl-passphrase-options.html

So you could write the password (including any emoji) to the first line of a file.
